spikeinterface/curation/curation_tools.py

```diff
@@ -33,6 +32,21 @@ def _find_duplicated_spikes_numpy(spike_train: np.ndarray, censored_period: int,
     return indices_of_duplicates
 
 
+def _find_duplicated_spikes_random(spike_train: np.ndarray, censored_period: int, seed: int) -> np.ndarray:
+    rand_state = np.random.get_state()
+    np.random.seed(seed)
+
+    indices_of_duplicates = []
+    while not np.all(np.diff(np.delete(spike_train, indices_of_duplicates)) > censored_period):
+        duplicates, = np.where(np.diff(spike_train) <= censored_period)
+        duplicates = np.unique(np.concatenate((duplicates, duplicates + 1)))
+        duplicate = np.random.choice(duplicates)
```

Comment:
`np.random.RandomState(seed=seed).choice()` this avoid to make variable for random state.

Reply:
done